hdfs/src/test/java/io/pravega/storage/hdfs/HDFSIntegrationTest.java

```diff
@@ -0,0 +1,83 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.storage.hdfs;
+
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageConfig;
+import io.pravega.segmentstore.server.host.BookKeeperIntegrationTestBase;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperConfig;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory;
+import lombok.val;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.junit.After;
+import org.junit.Before;
+
+/**
+ * End-to-end tests for SegmentStore, with integrated Storage and DurableDataLog.
+ */
+public class HDFSIntegrationTest extends BookKeeperIntegrationTestBase {
+    //region Test Configuration and Setup
+
+    private MiniDFSCluster hdfsCluster = null;
+
+    /**
+     * Starts BookKeeper and HDFS MiniCluster.
+     */
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        this.hdfsCluster = HDFSClusterHelpers.createMiniDFSCluster(getBaseDir().getAbsolutePath());
+        this.configBuilder.include(HDFSStorageConfig
+                .builder()
+                .with(HDFSStorageConfig.REPLICATION, 1)
+                .with(HDFSStorageConfig.URL, String.format("hdfs://localhost:%d/", hdfsCluster.getNameNodePort())));
+    }
+
+    /**
+     * Shuts down BookKeeper and HDFS MiniCluster.
+     */
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        val hdfs = this.hdfsCluster;
+        if (hdfs != null) {
+            hdfs.shutdown();
+            this.hdfsCluster = null;
+        }
+
+        super.tearDown();
+    }
+
+    //endregion
+
+    //region StreamSegmentStoreTestBase Implementation
+
+    @Override
+    protected boolean appendAfterMerging() {
+        return false; // HDFS is slow enough as it is; adding this would cause the test to take even longer.
+    }
+
+    @Override
+    protected ServiceBuilder createBuilder(ServiceBuilderConfig.Builder configBuilder, int instanceId, boolean useChunkedSegmentStorage) {
+        ServiceBuilderConfig builderConfig = getBuilderConfig(configBuilder, instanceId);
+        return ServiceBuilder
+                .newInMemoryBuilder(builderConfig)
+                .withStorageFactory(setup -> useChunkedSegmentStorage ?
+                        new HDFSSimpleStorageFactory(ChunkedSegmentStorageConfig.DEFAULT_CONFIG,
+                                setup.getConfig(HDFSStorageConfig::builder),
+                                setup.getStorageExecutor())
+                        : new HDFSStorageFactory(setup.getConfig(HDFSStorageConfig::builder), setup.getStorageExecutor()))
+                .withDataLogFactory(setup -> new BookKeeperLogFactory(setup.getConfig(BookKeeperConfig::builder), getBookkeeper().getZkClient(), setup.getCoreExecutor()));
+    }
+
+    //endregion
+}
```

Comment:
The libs does not appear in the branch. It only appears in the review. This is weird.

Reply:
https://github.com/pravega/pravega-storage-adapters/pull/20/commits/91d0faf7b427b88d87b2c8df08c595c7b1f7ab96 Adds lib
looks like https://github.com/pravega/pravega-storage-adapters/pull/20/commits/ae86e455706b7dc296d128b632c476c8650380d9 removes it